### PR TITLE
feat: add current HIP-24 spec

### LINKED ISSUE #9: Support HIP-24

### PR DESCRIPTION
**Pull request by janaakhterov**
Signed-off-by: Daniel Akhterov <daniel@launchbadge.com>

**Description**:

**Related issue(s)**:

Fixes #9 

**Notes for reviewer**:

**Checklist**

- [X] Documented (Code comments, README, etc.)

### CODE REVIEW

File: _sidebar.md [hunk 1]
```diff
@@ -111,6 +111,7 @@
         * [`TokenRevokeKycTransaction`](reference/token/TokenRevokeKycTransaction.md)
         * [`TokenTransfer`](reference/token/TokenTransfer.md)
         * [`TokenUnfreezeTransaction`](reference/token/TokenUnfreezeTransaction.md)
+        * [`TokenUnpauseTransaction`](reference/token/TokenUnpauseTransaction.md)
         * [`TokenUpdateTransaction`](reference/token/TokenUpdateTransaction.md)
```
Comment (SimiHunjan) [inline _sidebar.md:116]: We are missing `TokenPauseTransaction` from this list.

File: reference/token/TokenPauseTransaction.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+```javascript
+const transaction = new TokenPauseTransaction()
+    .setTokenId(tokenId);
+
+const response = await transaction.execute(client) // TransactionResponse;
+const receipt = await response.getReceipt(client) // TransactionReceipt;
+```
+
+- `Duration` is `number` and is the number of seconds
```
Comment (SimiHunjan) [inline reference/token/TokenPauseTransaction.md]: I think this line was accidentally added here.

File: reference/token/TokenUpdateTransaction.md [hunk 3]
```diff
@@ -153,3 +153,10 @@ The new account which will be automatically charged to renew the token's expirat
 The new interval at which the auto-renew account will be charged to extend the token's expiry.
 
 ---
+
```
Comment (SimiHunjan) [inline reference/token/TokenUpdateTransaction.md:156]: Should each of the examples be updated to show all of the fields that can be set?

File: reference/token/TokenCreateTransaction.md [hunk 4]
```diff
@@ -197,3 +197,10 @@ Note: This `AccountId` **MUST** sign the transaction if present.
 The interval at which the auto-renew account will be charged to extend the token's expiry
 
 ---
+
```
Comment (SimiHunjan) [inline reference/token/TokenCreateTransaction.md:200]: Should each of the examples be updated to show all of the fields that can be set?

**Review by SimiHunjan — CHANGES_REQUESTED** [review]
please see comments

File: reference/token/TokenUpdateTransaction.md (hunk 3 above)
Comment (janaakhterov) [reply reference/token/TokenUpdateTransaction.md:156]: The examples aren't meant to show you how to set _each_ field. They're simply supposed to show you the pattern of how to set fields in a particular language. For instance in Java to set fields you'd use `set<field name in title case>()` while in Go you'd do `Set<field name in title case>()`. The examples are meant to be simple enough that someone can look at them and get the gist of how to use the request type. This _can_ be changed, but that is the current goal of the examples which is why the examples weren't updated with `setPauseKey(pauseKey)`

**Review by SimiHunjan — APPROVED** [review]

**Comment by janaakhterov** [comment]
I discussed this a bit yesterday with @andrix10 and he had a good point. Why are we treating `TokenPauseStatus` differently than `TokenFreezeStatus` or `TokenKycStatus`. We we should update this PR to use a nullable boolean instead of an enum to be consistent with the other two statuses. Thoughts @Sean-Tedrow-LB @valtyr-naut

**Review by SimiHunjan — APPROVED** [review]

**Pull request merged**